openwisp_controller/config/models.py

```diff
@@ -99,6 +99,12 @@ class Meta(AbstractDevice.Meta):
         )
         abstract = False
 
+    def __str__(self):
+        if app_settings.HARDWARE_ID_ENABLED:
```

Comment:
I'd switch to a oneliner to avoid code coverage decreasing:

```python
return self.hardware_id if app_settings.HARDWARE_ID_ENABLED else self.name
```

Reply:
@nemesisdesign Thank you for the hint  - done! :grinning: